server/py/framework/utils/clients/iguazio/v4.py

```diff
@@ -26,6 +27,42 @@
 
 
 class Client(BaseClient):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self._client = iguazio.Client(api_url=self._api_url)
+
+    def refresh_access_token(
+        self, secret_token: mlrun.common.schemas.SecretToken
+    ) -> None:
+        """
+        Refreshes the access token by validating the provided offline token using the Iguazio client.
+
+        :param secret_token: SecretToken object containing the token name and the offline token string.
+        :raises mlrun.errors.MLRunUnauthorizedError: If the offline token is invalid or expired.
+        """
+        try:
+            self._client.refresh_access_token(secret_token.token)
+        except Exception as exc:
+            raise mlrun.errors.MLRunUnauthorizedError(
```

Comment:
What if another exception occurred (e.g. timeout) - do we still return Unauthorized?